doc/development.rst

```diff
@@ -192,6 +192,56 @@ Upon a new release, the PEtab editors ensure that
 * the new version of the specifications is deposited at Zenodo
 * the new release is announced on the PEtab mailing list
 
+PEtab Extensions
+----------------
+
+An elaborate, monolithic format would make it difficult to understand and
+implement support for PEtab, leading to a steep learning curve and discouraging
+support in new toolboxes. To address this issue, the PEtab format is modular and
+permits modifications through extensions that complement the core standard.
+This modular specification evens the learning curve and provides toolbox
+developers with more guidance on which features to implement to maximize
+support for real world applications. Moreover, such modular extensions
+facilitate and promote the use of specialized tools for specific, non-parameter
+estimation tasks such as visualization.
+
+Requirements for new extensions:
+
+* Specifications in PEtab extensions take precedence over PEtab core, i.e., they
+  can ease or refine format restrictions imposed by PEtab core.
+* PEtab extensions should extend PEtab core with new orthogonal features or
+  tasks, i.e., they should not make trivial changes to PEtab core.
+* PEtab extensions must be named according to ^[a-zA-Z][\w\-]*$
+* PEtab extensions must be versioned using semantic versioning.
+* PEtab extensions required for interpretation of a problem specification must
+  be specified in the PEtab-YAML files
+* There is at least one tool that supports the proposed extension
+* The authors provide a library that provides test cases and implements
+  validation checks for the proposed format.
```

Comment:
I'd like to keep the library out of the specification, but it should be added to the library documentation.

Reply:
Makes sense, since this document is meant to be static. I guess the statement advising community engagement covers this. That could be broadened to "It is encouraged that (potential) extensions are informally discussed with the community as early as possible..."